Compactable/Type

```diff
@@ -0,0 +1,19 @@
+{-
+
+Compactable serves as an abstraction over filtering and partitioning, using two functions
+compact and separate, respectively.
```

Comment:
This makes it sound like the type class still has multiple methods. I would rewrite this to discuss `mapOptional` instead, and then move the other comments to the respective expressions.